### ISSUE TITLE
Instructions Unclear for installing SDL2

### ISSUE THREAD

**Issue by marcus13345** [kmamal/node-sdl#1]
SDL2 has no installer, and it is unclear where im meant to put the `SDL2/lib` and `SDL2/include` directories to allow node-gyp to build correctly.

Edit:
Per the new documentation, ive added `SDL2.dll` to `C:\Windows` (also tried `C:\Windows\System32`, and am still receiving the same build error:

```
yarn add v1.6.0
info No lockfile found.
[1/4] Resolving packages...
warning @kmamal/sdl > prebuild > cmake-js > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142 warning @kmamal/sdl > prebuild > node-ninja > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @kmamal/sdl > prebuild > nw-gyp > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @kmamal/sdl > prebuild > node-gyp > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142 warning @kmamal/sdl > prebuild > cmake-js > request > har-validator@5.1.5: this library is no longer supported
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
[1/1] ⠁ @kmamal/sdl: ..\src\sdl-helpers.cpp(2): fatal error C1083: Cannot open include file: 'SDL.h': No such file or directory [C:
[-/1] ⠁ waiting...
[-/1] ⠁ waiting...
[-/1] ⠁ waiting...
error C:\Users\Marcus\sdl\node_modules\@kmamal\sdl: Command failed.
Exit code: 1
Command: prebuild-install || node-gyp rebuild
Arguments:
Directory: C:\Users\Marcus\sdl\node_modules\@kmamal\sdl
Output:
'prebuild-install' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\Marcus\sdl\node_modules\@kmamal\sdl>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@16.1.0 | win32 | x64
gyp info find Python using Python version 3.9.5 found at "C:\Users\Marcus\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\python.exe"
gyp info find VS using VS2017 (15.5.27130.0) found at:
gyp info find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community"
gyp info find VS run with --verbose for detailed information
(node:1154796) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
(Use `node --trace-deprecation ...` to show where the warning was created)
gyp info spawn C:\Users\Marcus\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\python.exe
gyp info spawn args [
gyp info spawn args   'C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\Marcus\\sdl\\node_modules\\@kmamal\\sdl\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\Marcus\\AppData\\Local\\node-gyp\\Cache\\16.1.0\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\Marcus\\AppData\\Local\\node-gyp\\Cache\\16.1.0',
gyp info spawn args   '-Dnode_gyp_dir=C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Users\\\\Marcus\\\\AppData\\\\Local\\\\node-gyp\\\\Cache\\\\16.1.0\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\Users\\Marcus\\sdl\\node_modules\\@kmamal\\sdl',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\Users\\Marcus\\sdl\\node_modules\\@kmamal\\sdl\\build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
Warning: unrecognized setting ClCompile/LanguageStandard
Warning: unrecognized setting ClCompile/LanguageStandard
gyp info spawn C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe
gyp info spawn args [
gyp info spawn args   'build/binding.sln',
gyp info spawn args   '/clp:Verbosity=minimal',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=x64'
gyp info spawn args ]
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  binding.cpp
  sdl-helpers.cpp
..\src\sdl-helpers.cpp(2): fatal error C1083: Cannot open include file: 'SDL.h': No such file or directory [C:\Users\Marcus\sdl\node_modules\@kmamal\sdl\build\test.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1 gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Windows_NT 10.0.18363
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\Marcus\sdl\node_modules\@kmamal\sdl
```

**Comment by kmamal** [comment]
Just published `v0.0.11` to npm. If you're on x64 windows there should be prebuilt binaries so no need to build anything.

**Comment by marcus13345** [comment]
Just in case there is any complications, I did the following:
- downgraded node to current LTS `14.17.0`
- created a new project
- switched to npm over yarn
- `npm init`
- `npm i @kmamal/sdl`

and i still receive the same (or very similar...) error:

[here is a pastebin with the full npm error log](https://pastebin.com/raw/itraRczQ)

Any additional information you may want, I'm happy to provide!

**Comment by kmamal** [comment]
I think the arguments to `prebuild-install` were wrong. I was missing `--runtime napi` and so it went looking for a node version which I don't build. Published `v0.0.12`. Try that one.

**Comment by kmamal** [comment]
Found some more issues on windows. Published `v0.0.13`. As you've discovered this lib is still rather unstable so please let me know about any problems you're having. Thanks for the feedback!

**Comment by marcus13345** [comment]
of course! please, feel free to email me sometime `marcus@valnet.xyz`. I've been in search of a good SDL library in node for a while now, and I'm really excited to finally be able to do it!
